### ISSUE TITLE
Remove (useless) and (almost) unused legacy PHP-to-JS variables

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#3759]
Follow-up to #3647 and #3758: A bunch of useless legacy JS globals provided from PHP should be removed.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* Any remainders of `_googlesitekitLegacyData.admin.datacache` that may still be present after #2258 (including server-side) should be entirely removed, since no longer relevant with the new JS API layer.
* The `_googlesitekitLegacyData.admin.nojscache` variable and all usages (in JS and PHP) should be removed - this technically removes the ability to bypass cache, but that shouldn't be necessary to use anyway.
* The following other unused legacy global variables should be entirely removed as well:
    * `_googlesitekitLegacyData.admin.siteName`
    * `_googlesitekitLegacyData.admin.siteUserID`
    * `_googlesitekitLegacyData.admin.assetsRoot`
    * `_googlesitekitLegacyData.admin.timestamp`
    * `_googlesitekitLegacyData.admin.currentScreen`
    * `_googlesitekitLegacyData.admin.currentAdminPage`
    * `_googlesitekitLegacyData.admin.ampEnabled`
    * `_googlesitekitLegacyData.admin.ampMode`
    * `_googlesitekitLegacyData.admin.homeURL`
    * `_googlesitekitLegacyData.notifications`
    * `_googlesitekitLegacyData.permaLink`
    * `_googlesitekitLegacyData.pageTitle`
    * `_googlesitekitLegacyData.publicPath`

## Implementation Brief

* 

### Test Coverage

* 

### Visual Regression Changes

* 

## QA Brief

* 

## Changelog entry

*

**Comment by eugene-manuilov** [comment]
IB :heavy_check_mark:

**Comment by felixarntz** [comment]
Note that the IB is here is technically slightly outdated: Since this issue is blocked by #2258, some of the IB references of things to remove will already no longer exist.

**Comment by ivankruchkoff** [comment]
QA: ✅

No references to variables in AC in codebase.
